saltproc/saltproc.py

```diff
@@ -265,108 +500,166 @@ def write_mat_file(self):
-        # every 1 step = 3 days
-        volatile_gases = self.find_iso_indx(['Kr', 'Xe', 'Se', 'Nb', 'Mo', 'Tc', 'Ru',
-                                             'Rh', 'Pd', 'Ag', 'Sb', 'Te'])
-        self.rem_adens[0, ] = self.remove_iso(volatile_gases, 1)
-
-        # !! this rem_adens indexing looks wrong
-        # remove seminoble metals
-        # every 67 steps = 201 days
-        if self.current_step % 67 == 0:
-            se_noble_id = self.find_iso_indx(['Zr', 'Cd', 'In', 'Sn'])
-            self.rem_adens[1, ] = self.remove_iso(se_noble_id, 1)
-
-        # remove volatile fluorides
-        # every 20 steps = 60 days
-        if self.current_step % 20 == 0:
-            vol_fluorides = self.find_iso_indx(['Br', 'I'])
-            self.rem_adens[2, ] = self.remove_iso(vol_fluorides, 1)
-
-        # remove REEs
-        # evrey 17 steps = 50 days
-        if self.current_step % 17 == 0:
-            rees_id = self.find_iso_indx(['Y', 'Gd', 'La', 'Ce', 'Pr',
-                                          'Nd', 'Pm', 'Sm'])
-            self.rem_adens[3, ] = self.remove_iso(rees_id, 1)
-
-        # remove Eu
-        # evrey 167 steps = 500 days
-        if self.current_step % 167 == 0:
-            eu_id = self.find_iso_indx('Eu')
-            self.rem_adens[4, ] = self.remove_iso(eu_id, 1)
-
-        # remove Rb, Sr, Cs, Ba
-        # every 1145 steps = 3435 days
-        if self.current_step % 1145 == 0:
-            discard_id = self.find_iso_indx(['Rb', 'Sr', 'Cs', 'Ba'])
-            self.rem_adens[4, ] = self.remove_iso(discard_id, 1)
-
-        # remove np-237, pu-242
-        # every 1946 steps = 16 years
-        if self.current_step % 1946 == 0:
-            higher_nuc = self.find_iso_indx(['Pu237', 'Pu242'])
-            self.rem_adens[4, ] = self.remove_iso(higher_nuc, 1)
-
-        # refill th232 to keep adens constant
-        # do it every time
-        # if want to do it less often do:
-        # if current_step % time == 0:
-        self.th_adens_db[self.current_step, ] = self.maintain_const(th232_id,
-                                                                    self.th232_adens_0)
-
-        # write the processed material to mat file for next run
-        self.write_mat_file()
+        self.core = self.read_dep()
+
+        # make core values mass by multiplying by volume
+        self.core[self.driver_mat_name] = self.core[self.driver_mat_name] * \
+            self.driver_vol
+        try:
+            self.core[self.blanket_mat_name] = self.core[self.blanket_mat_name] * \
+                self.blanket_vol
+        except:
+            self.core[self.blanket_mat_name] = np.zeros(
+                self.number_of_isotopes)
+        # save pre-processing core mass
+        self.core_mass = {}
+        for key, val in self.core.items():
+            self.core_mass[key] = sum(val)
+
+        # record the depleted composition before reprocessing
+        self.driver_before_db[self.current_step,
```

Comment:
same comments as before. I want to see driver and blanket irradicated from the general behavior of salt proc. Saltproc should be designed to be geometry agnostic (and certainly should continue to be useable for single fluid reactors). Geometry specifics can be handled elsewhere (a geometry class, reactor class, what-ever) where the salt proc methods can loop through reactor fluids as the geometry requires.